src/box/lua/config/instance_config.lua

```diff
@@ -2311,7 +2311,24 @@ return schema.new('instance_config', schema.record({
                 type = 'number',
                 default = 10,
             }),
-        })
+        }),
+        replicasets = schema.map({
+            -- Name of the replica.
+            key = schema.scalar({
+                type = 'string',
+            }),
+            value = schema.record({
+                -- Priorities for the supervised failover mode.
+                priority = enterprise_edition(schema.map({
```

Comment:
The whole `failover` section relates to the enterprise edition. I wouldn't add a check specifically for this option.

Proposed changes (together with ones for the new `luatest.treegen` helper):

```diff
diff --git a/src/box/lua/config/instance_config.lua b/src/box/lua/config/instance_config.lua
index ea91e48d79..424164ad12 100644
--- a/src/box/lua/config/instance_config.lua
+++ b/src/box/lua/config/instance_config.lua
@@ -2318,14 +2318,14 @@ return schema.new('instance_config', schema.record({
             }),
             value = schema.record({
                 -- Priorities for the supervised failover mode.
-                priority = enterprise_edition(schema.map({
+                priority = schema.map({
                     key = schema.scalar({
                         type = 'string',
                     }),
                     value = schema.scalar({
                         type = 'number',
                     }),
-                })),
+                }),
             }),
         }),
     }),
diff --git a/test/config-luatest/config_test.lua b/test/config-luatest/config_test.lua
index 432986733a..52d7c1548f 100644
--- a/test/config-luatest/config_test.lua
+++ b/test/config-luatest/config_test.lua
@@ -922,10 +922,8 @@ g.test_failover_supervised_constrainsts = function()
     })
 end
 
-g.test_failover_config = function(g)
-    t.tarantool.skip_if_not_enterprise()
-
-    local dir = treegen.prepare_directory(g, {}, {})
+g.test_failover_config = function()
+    local dir = treegen.prepare_directory({}, {})
     local config = [[
         credentials:
           users:
@@ -950,7 +948,7 @@ g.test_failover_config = function(g)
                   instance-002: {}
                   instance-003: {}
     ]]
-    local config_file = treegen.write_script(dir, 'config.yaml', config)
+    local config_file = treegen.write_file(dir, 'config.yaml', config)
     local env = {TT_LOG_LEVEL = 0}
     local args = {'--name', 'instance-001', '--config', config_file}
     local opts = {nojson = true, stderr = true}
@@ -988,7 +986,7 @@ g.test_failover_config = function(g)
                   instance-002: {}
                   instance-003: {}
     ]]
-    local config_file = treegen.write_script(dir, 'config.yaml', config)
+    local config_file = treegen.write_file(dir, 'config.yaml', config)
     local env = {TT_LOG_LEVEL = 0}
     local args = {'--name', 'instance-001', '--config', config_file}
     local opts = {nojson = true, stderr = true}
```

Reply:
Applied the patch, thank you!